node/beacon/src/authority.rs

```diff
@@ -440,4 +507,29 @@ mod test {
         assert_eq!(find_threshold(&[1000, 1, 1, 1, 1, 1, 1, 1, 1, 1], 1).unwrap(), 1000);
         assert!(find_threshold(&[1, 1, 2], 100).is_err());
     }
+
+    #[test]
+    fn test_write_to_storage() {
+        let chain_spec = get_test_chainspec(4, 2, 2);
+        let bc = test_blockchain(0, &chain_spec);
+        let mut authority = bc.authority.write().unwrap();
+        let block1 = SignedBeaconBlock::new(1, bc.chain.genesis_hash(), vec![], CryptoHash::default());
+        let mut header1 = block1.header();
+        header1.signature.authority_mask = vec![true, true];
+        let block2 = SignedBeaconBlock::new(2, header1.block_hash(), vec![], CryptoHash::default());
+        let mut header2 = block2.header();
+        header2.signature.authority_mask = vec![true, true];
+        authority.process_block_header(&header1);
+        authority.process_block_header(&header2);
+        let next_authorities = authority.get_authorities(3);
+        assert!(next_authorities.is_ok());
+
+        let bc1 = BeaconClient::new(
+            block2.clone(),
```

Comment:
isn't this a genesis block as the first argument and you are passing block with `index=2`? how is this not failing - it's offtopic to this PR, but let's add a check that if BeaconClient started with different genesis, it should fail.

Reply:
Created an issue for this https://github.com/nearprotocol/nearcore/issues/690